src/Explorer/Controls/SmartUi/SmartUiComponent.tsx

```diff
@@ -101,42 +116,167 @@ export class SmartUiComponent extends React.Component<SmartUiComponentProps, Sma
+        console.log(this.props.descriptor.inputNames);
+        this.setState({ isRefreshing: false });
+        throw new Error(`${key} is not an input property of this class.`);
+      }
+      currentValues = currentValues.set(key, initialValues.get(key));
+      baselineValues = baselineValues.set(key, initialValues.get(key));
+    }
+    this.setState({ currentValues: currentValues, baselineValues: baselineValues, isRefreshing: false });
+  };
+
+  private discard = (): void => {
+    let { currentValues } = this.state;
+    const { baselineValues } = this.state;
+    for (const key of baselineValues.keys()) {
+      currentValues = currentValues.set(key, baselineValues.get(key));
+    }
+    this.setState({ currentValues: currentValues });
+  };
+
+  private setDefaults = async (currentNode: Node): Promise<void> => {
+    if (currentNode.info && currentNode.info instanceof Function) {
+      currentNode.info = await (currentNode.info as infoPromise)();
+    }
+
+    if (currentNode.input) {
+      currentNode.input = await this.getModifiedInput(currentNode.input);
+    }
+    const promises = currentNode.children?.map(async (child: Node) => await this.setDefaults(child));
+    if (promises) {
+      await Promise.all(promises);
+    }
+  };
+
+  private getModifiedInput = async (input: AnyInput): Promise<AnyInput> => {
+    if (input.label instanceof Function) {
+      input.label = await (input.label as stringPromise)();
+    }
+
+    if (input.placeholder instanceof Function) {
+      input.placeholder = await (input.placeholder as stringPromise)();
+    }
+
+    switch (input.type) {
+      case "string": {
+        return input as StringInput;
+      }
+      case "number": {
+        const numberInput = input as NumberInput;
+        if (numberInput.min instanceof Function) {
+          numberInput.min = await (numberInput.min as numberPromise)();
+        }
+        if (numberInput.max instanceof Function) {
+          numberInput.max = await (numberInput.max as numberPromise)();
+        }
+        if (numberInput.step instanceof Function) {
+          numberInput.step = await (numberInput.step as numberPromise)();
+        }
+        return numberInput;
+      }
+      case "boolean": {
+        const booleanInput = input as BooleanInput;
+        if (booleanInput.trueLabel instanceof Function) {
+          booleanInput.trueLabel = await (booleanInput.trueLabel as stringPromise)();
+        }
+        if (booleanInput.falseLabel instanceof Function) {
+          booleanInput.falseLabel = await (booleanInput.falseLabel as stringPromise)();
+        }
+        return booleanInput;
+      }
+      default: {
+        const enumInput = input as DropdownInput;
+        if (enumInput.choices instanceof Function) {
+          enumInput.choices = await (enumInput.choices as dropdownItemPromise)();
+        }
+        return enumInput;
+      }
+    }
+  };
+
+  private getDefaultValue = (input: AnyInput): InputType => {
```

Comment:
>getDefaultValue [](start = 10, length = 15)

unused.. please remove

Reply:
@srinathnarayanan any unused stuff should be getting flagged by the linter. Was this function not?

Reply:
ya it wasn't. Removing it.